src/ol/render/canvas/Executor.js

```diff
@@ -363,12 +368,10 @@ class Executor {
       p4[1] = p3[1];
     }
 
-    let transform = null;
+    let transform;
     if (rotation !== 0) {
-      const centerX = x + anchorX;
-      const centerY = y + anchorY;
       transform = composeTransform(
-        tmpTransform,
+        createTransform(),
```

Comment:
That's because we store the arguments to `replayImageOrLabel` for the image style until we render the according text style. If we'd continue to use `tmpTransform`, the transform would change before we call `replayImageOrLabel`.